src/anomalib/utils/metrics/aupro.py

```diff
@@ -96,9 +108,12 @@ def compute_pro(self, cca: Tensor, target: Tensor, preds: Tensor) -> tuple[Tenso
         Returns:
             tuple[Tensor, Tensor]: tuple containing final fpr and tpr values.
         """
+        # initialize the roc curve with the specified thresholds
+        # target being forced to be 0 or 1, we can use the binary roc curve
+        roc_in_pro = partial(roc, task="binary", thresholds=self.thresholds)
```

Comment:
- roc to binary_roc: indeed and will do !

- partial usage: theroc with thresholds as argument is used twice. So to avoid to forgot the thresholds in the second call I decided to go with the partial trick. @jpcbertoldo If you are not comfortable with it, I can use directly:
 `binary_roc(preds, target, thresholds=self.thresholds)`

What is your last call ?

Reply:
I see the point, but since the two calls are so close I believe simpler is better (I would agree with what you said if there were many calls, or a long snippet, or passing the function as an argument).

Reply:
done